### ISSUE TITLE
spacy.cli.download is no longer available

### ISSUE THREAD

**Issue by omri374** [explosion/spaCy#13039]
In the last release (3.7.0), the ability to call `spacy.cli.download(MODEL)` is no longer available, which is backward incompatible.
Is there another way to download models through a Python script, or can the spacy.cli package be reintroduced?

Potentially an outcome of #12962

## How to reproduce the behaviour
```python
import spacy

spacy.cli.download("en_core_web_lg")

# AttributeError: module 'spacy' has no attribute 'cli'
```
## Your Environment

- **spaCy version:** 3.7.0
- **Platform:** macOS-14.0-arm64-arm-64bit
- **Python version:** 3.9.16
- **Pipelines:** el_core_news_lg (3.5.0), en_core_web_lg (3.6.0), sv_core_news_sm (3.6.0), en_core_web_sm (3.6.0), en_core_web_trf (3.5.0), es_core_news_sm (3.5.0), es_core_news_md (3.5.0), en_core_web_md (3.5.0)

**Comment by adrianeboyd** [comment]
That does look like an unintended side effect of #12962. My initial guess is that the best course of action will be to revert it.

**Comment by omri374** [comment]
Update: calling `from spacy.cli import download` works. It's just the specific availability of `spacy.cli` which had changed.

**Comment by adrianeboyd** [comment]
Yeah, we have do have plenty of tests including imports like `from spacy.cli import download`, but not for just `spacy.cli` or `spacy.cli.download`.

**Comment by adrianeboyd** [comment]
And I don't understand why this breaks the availability of the whole `spacy.cli` module. But it's pretty breaking, so we'll release v3.7.1 with this reverted.
